lib/config/helper.go

```diff
@@ -0,0 +1,328 @@
+	}
+}
+
+func checkForS3BackupConfig(c *HaDeployConfig, errorList *list.List) {
+	if c.Architecture.Aws.BackupConfig == "s3" {
+		checkForValidS3Bucket(c.Architecture.Aws, errorList)
+	}
+}
+
+func checkManagedServicesBackupConfig(c *HaDeployConfig, errorList *list.List) {
+	if c.Architecture.Aws.BackupConfig != "s3" {
+		errorList.PushBack("Invalid backup_config. Only 's3' is supported.")
+	}
+}
+
+func checkNonManagedServicesBackupConfig(c *HaDeployConfig, errorList *list.List) {
+	if c.Architecture.Aws.BackupConfig != "efs" && c.Architecture.Aws.BackupConfig != "s3" {
+		errorList.PushBack("Invalid backup_config. It should be 'efs' or 's3'.")
+	}
+}
+
+func checkForValidS3Bucket(configInitials *ConfigInitials, errorList *list.List) {
+	if configInitials.BackupConfig == "s3" && len(strings.TrimSpace(configInitials.S3BucketName)) < 1 {
+		errorList.PushBack("Invalid or empty s3_bucketName.")
+	}
+}
+
+func validateAutomateAdminPassword(automateSettings *ConfigAutomateSettings, errorList *list.List) {
+	if len(automateSettings.AdminPassword) > 0 {
+		val, err := password.NewValidator()
+		if err != nil {
+			errorList.PushBack(err.Error())
+		}
+		passvalErr := val.Validate(automateSettings.AdminPassword)
+		if passvalErr != nil {
+			errorList.PushBack(passvalErr.Error())
+		}
+	}
+}
+
+func validateAutomateCerts(automateSettings *ConfigAutomateSettings, errorList *list.List) {
+	if len(strings.TrimSpace(automateSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(automateSettings.PublicKey)) < 1 {
+		errorList.PushBack("Automate public_key and/or private_key are missing. Otherwise set enable_custom_certs to false.")
+	}
+
+	if len(strings.TrimSpace(automateSettings.RootCA)) > 0 {
+		errorList.PushBackList(checkCertValid([]keydetails{
+			{key: automateSettings.RootCA, certtype: rootCa, svc: automate},
+		}))
+	}
+
+	errorList.PushBackList(checkCertValid([]keydetails{
+		{key: automateSettings.PrivateKey, certtype: privateKey, svc: automate},
+		{key: automateSettings.PublicKey, certtype: publicKey, svc: automate},
+	}))
+}
+
+func validateChefServerCerts(chefServerSettings *ConfigSettings, errorList *list.List) {
+	if len(strings.TrimSpace(chefServerSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(chefServerSettings.PublicKey)) < 1 {
+		errorList.PushBack("ChefServer root_ca and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false.")
+	}
+
+	errorList.PushBackList(checkCertValid([]keydetails{
+		{key: chefServerSettings.PrivateKey, certtype: privateKey, svc: chefServer},
+		{key: chefServerSettings.PublicKey, certtype: publicKey, svc: chefServer},
+	}))
+}
+
+func checkCertValid(keys []keydetails) *list.List {
+	errorList := list.New()
+	for _, el := range keys {
+		block, _ := pem.Decode([]byte(el.key))
+		if block == nil {
+			errorList.PushBack("Invalid format. Failed to decode " + el.certtype + " for " + el.svc)
+		} else {
+			switch el.certtype {
```

Comment:
In case if it is not a rootCA why are we adding error?

Reply:
updated code, to avoid that